internal/controllers/utils/constants.go

```diff
@@ -78,15 +78,23 @@ const (
 )
 
 var (
-	// Non-immutable ClusterInstance fields at the cluster-level
-	AllowedClusterInstanceClusterFields = []string{
-		"extraAnnotations",
-		"extraLabels",
+	// AllowedClusterInstanceFields contains path patterns for fields that are allowed to be updated.
+	AllowedClusterInstanceFields = [][]string{
+		// Cluster-level non-immutable fields
+		{"extraAnnotations"},
+		{"extraLabels"},
+		// Node-level non-immutable fields
+		{"nodes", "*", "extraAnnotations"},
+		{"nodes", "*", "extraLabels"},
 	}
-	// Non-immutable ClusterInstance fields at the node-level
-	AllowedClusterInstanceNodeFields = []string{
-		"extraAnnotations",
-		"extraLabels",
+
+	// IgnoredClusterInstanceFields contains path patterns for fields that should be ignored.
+	IgnoredClusterInstanceFields = [][]string{
+		// Node-level ignored fields
+		{"nodes", "*", "bmcAddress"},
+		{"nodes", "*", "bmcCredentialsName"},
+		{"nodes", "*", "bootMACAddress"},
+		{"nodes", "*", "nodeNetwork", "interfaces", "*", "macAddress"},
```

Comment:
Why is this listing specific fields but also including a wildcard?

Reply:
I think the wildcard here is an index into a list of X ... ie. to match node[0].bmcAddress, for example. Probably would be good to comment that here?

Reply:
Updated with a comment. The wildcard in that specific position represents the index in a list.

Reply:
> I think the wildcard here is an index into a list of X ... ie. to match node[0].bmcAddress, for example. Probably would be good to comment that here?

Yep, added a comment